### PR TITLE
Fix Wedge and AnnualWedge glyph hitboxes

### LINKED ISSUE #12070: [BUG] Wedge/AnnularWedge hit testing cuts off top/bottom
### Software versions

Tested on both 2.2.3 and the latest "branch-3.0" development branch.

Python version      :  3.8.13 | packaged by conda-forge | (default, Mar 25 2022, 06:04:10) 
IPython version     :  8.2.0
Tornado version     :  6.1
Bokeh version       :  3.0.0dev5+23.g6f1e39a2a.dirty
BokehJS static path :  /home/msbrown/bkdev/bokeh/bokeh/server/static
node.js version     :  v16.14.2
npm version         :  8.5.0
Operating system    :  Linux-5.13.0-27-generic-x86_64-with-glibc2.10

### Browser name and version

_No response_

### Jupyter notebook / Jupyter Lab version

_No response_

### Expected behavior

The Wedge and AnnularWedge glyphs should consider the whole height of the circle for hit testing, including the top and bottom.

### Observed behavior

Depending on the X and Y scale, the tops and bottoms of these glyphs are cut off for hit-testing purposes. The user cannot click to select or hover over these parts.

The amount that is cut off depends on the difference in scale between the X and Y axes. This is because these glyphs are drawn as parts of circles in _screen_ space, not in data space. Because of this, the rectangle given to the index (`this.index.indices({x0, x1, y0, y1})`) is incorrect.

### Example code

```Python
See the [pie chart example](https://docs.bokeh.org/en/latest/docs/gallery/pie_chart.html) in the gallery.
```

### Stack traceback or browser console output

_No response_

### Screenshots

Screenshot_2022-04-01_11-16-23

### PR DESCRIPTION
**Pull request by mbrown1413**
All pull requests must have an associated issue in the issue tracker. If there
isn't one, please go open an issue describing the defect, deficiency or desired
feature. You can read more about our issue and PR processes in the
[wiki](https://github.com/bokeh/bokeh/wiki/BEP-1:-Issues-and-PRs-management).

- [x] issues: fixes #12070
- [x] tests added / passed
- [ ] release document entry (if new feature or API change)

This changes the hit-test calculations in Wedge and AnnularWedge so they are performed in screen-space, since the glyphs are rendered as circles in screen space.

This doesn't add any tests yet, but I'll work on that soon.

### CODE REVIEW

File: bokehjs/src/lib/models/glyphs/wedge.ts [hunk 1]
```diff
@@ -58,28 +59,19 @@ export class WedgeView extends XYGlyphView {
...
-    } else {
-      sx0 = sx - max_diameter
-      sx1 = sx + max_diameter
-      ;[x0, x1] = this.renderer.xscale.r_invert(sx0, sx1)
-
-      sy0 = sy - max_diameter
-      sy1 = sy + max_diameter
-      ;[y0, y1] = this.renderer.yscale.r_invert(sy0, sy1)
-    }
+    const s_max_radius = max(this.sradius)
```
Comment (mattpap) [inline bokehjs/src/lib/models/glyphs/wedge.ts:68]: Expensive computations, like this one, should be moved to `_map_data()` (the name should be `max_sradius`).

Comment (mbrown1413) [reply bokehjs/src/lib/models/glyphs/wedge.ts]: Makes sense. Pushed a change to this and a corresponding change in annular_wedge.ts

**Comment by mbrown1413** [comment]
I assume I'll need to write unit tests but not integration tests for this? Is `bokeh/bokehjs/test/unit/models/glyphs/rect.ts` a good starting point to copy? That file has a few assertions for hit-testing, and I don't see any specific unit tests for Wedge and AnnularWedge yet, so I'm thinking of copying it to `wedge.ts` / `annular_wedge.ts`.

Are the checks that have already failed due to some intermittent error? They look unrelated to my changes.

**Comment by mbrown1413** [comment]
I just pushed a new commit with tests. I wasn't entirely sure how to structure them, so let me know if I have anything wrong there. I checked locally that they fail without the fix and pass with the fix.

I think everything is ready in this PR. Let me know if there's something more I should do before it's reviewed.

**Review by bryevdv — APPROVED** [review]

**Comment by bryevdv** [comment]
Thanks @mbrown1413 !

**Pull request merged**